pkg/segment/writer/segwriter.go

```diff
@@ -706,7 +711,10 @@ func writeWip(colWip *ColWip, encType []byte) (uint32, int64, error) {
 func compressWip(colWip *ColWip, encType []byte) ([]byte, uint32, error) {
 	var compressed []byte
 	if bytes.Equal(encType, ZSTD_COMLUNAR_BLOCK) {
-		compressed = encoder.EncodeAll(colWip.cbuf[0:colWip.cbufidx], make([]byte, 0, colWip.cbufidx))
+
+		// reduce the len to 0, but keep the cap of the underlying buffer
+		compressed = encoder.EncodeAll(colWip.cbuf[0:colWip.cbufidx],
+			colWip.workBufForCompression[:0])
```

Comment:
Just a note: from the [documentation](https://pkg.go.dev/github.com/klauspost/compress/zstd#Encoder.EncodeAll), the second parameter (`workBufForCompression`) isn't just used as a workspace, it's where the encoded bytes are stored. After this call completes, `compressed` and `workBufForCompression` are the same